src/osd/osd_types.h

```diff
@@ -549,6 +556,15 @@ struct spg_t {
     }
     return is_split;
   }
+  bool is_merge(unsigned old_pg_num, unsigned new_pg_num, spg_t *parent) const {
+    spg_t out = *this;
```

Comment:
It looks like you could get away with a refernce, rather than a copy, but maybe there are side-effects I'm not noticing.